### DISCUSSION TITLE
Question about the generation of the incoherent Qz vs. E ellipse

### DISCUSSION THREAD

**Discussion by bingli621** [ILLGrenoble/takin#23] [Q&A]
I am having a hard time understanding how the incoherent ellipses in the Qz vs E is generated. I am looking at the code from this link (https://github.com/ILLGrenoble/takin/blob/master/core/tools/res_py/reso.py) from line 192 to 216. I've pasted the code below.
I can understand Qz is treated differently, and it can be deleted instead of projected out. But when the Qz vs. E ellipse is created, Q_ortho is deleted instead of projected. Is this correct?

# -------------------------------------------------------------------------------------------------------------------------------
# 2d projected ellipses
    if verbose:
        print()
    Qres_QxE_proj = np.delete(np.delete(Qres_Q, 2, axis=0), 2, axis=1)
    Qres_QxE_proj = quadric_proj(Qres_QxE_proj, 1)
    [fwhms_QxE_proj, angles_QxE_proj, rot_QxE_proj, evals_QxE_proj] = descr_ellipse(Qres_QxE_proj)
    if verbose:
        print("2d Qx,E projection fwhms and slope angle: %s, %.4f" % (fwhms_QxE_proj, angles_QxE_proj[0]))

    Qres_QyE_proj = np.delete(np.delete(Qres_Q, 2, axis=0), 2, axis=1)
    Qres_QyE_proj = quadric_proj(Qres_QyE_proj, 0)
    [fwhms_QyE_proj, angles_QyE_proj, rot_QyE_proj, evals_QyE_proj] = descr_ellipse(Qres_QyE_proj)
    if verbose:
        print("2d Qy,E projection fwhms and slope angle: %s, %.4f" % (fwhms_QyE_proj, angles_QyE_proj[0]))

    Qres_QzE_proj = np.delete(np.delete(Qres_Q, 1, axis=0), 1, axis=1)
    Qres_QzE_proj = quadric_proj(Qres_QzE_proj, 0)
    [fwhms_QzE_proj, angles_QzE_proj, rot_QzE_proj, evals_QzE_proj] = descr_ellipse(Qres_QzE_proj)
    if verbose:
        print("2d Qz,E projection fwhms and slope angle: %s, %.4f" % (fwhms_QzE_proj, angles_QzE_proj[0]))

    Qres_QxQy_proj = quadric_proj(Qres_Q, 3)
    Qres_QxQy_proj = np.delete(np.delete(Qres_QxQy_proj, 2, axis=0), 2, axis=1)
    [fwhms_QxQy_proj, angles_QxQy_proj, rot_QxQy_proj, evals_QxQy_proj] = descr_ellipse(Qres_QxQy_proj)
    if verbose:
        print("2d Qx,Qy projection fwhms and slope angle: %s, %.4f" % (fwhms_QxQy_proj, angles_QxQy_proj[0]))

**Comment by tweber-ill** [comment]
Hello,

yes, there's several different ways to visualise a four-dimensional ellipsoid via its two-dimensional slices and projections. In the main program, you can configure which axes you want to delete or project (File -> Preferences -> Resolution). In the _Python_ code I used this as the default configuration, but you can also modify it to -- for example -- delete the Q_para axis and project onto the Q_perp axis in the Qres_QzE_proj case instead.
